Sources/SwiftCoAP/SCClient.swift

```diff
@@ -11,23 +11,23 @@ import UIKit
 //MARK:
 //MARK: SC Client Delegate Protocol declaration
 
-@objc protocol SCClientDelegate {
+public protocol SCClientDelegate: AnyObject {
     
     //Tells the delegate that a valid CoAP message was received
     func swiftCoapClient(_ client: SCClient, didReceiveMessage message: SCMessage)
     
     //Tells the delegate that an error occured during or before transmission (refer to the "SCClientErrorCode" Enum)
-    @objc optional func swiftCoapClient(_ client: SCClient, didFailWithError error: NSError)
+    func swiftCoapClient(_ client: SCClient, didFailWithError error: NSError)
```

Comment:
We can provide a default implementation (later) so that the methods are still effectively “optional.”

Reply:
I've drilled through the code again. Personally I can't see the problem in requiring this methods explicitly to comply to the protocol. But hiding it with the default implementation (in this case it will be just an empty call) is seems less obvious.